iceoryx_hoofs/include/iceoryx_hoofs/cxx/optional.hpp

```diff
@@ -83,6 +89,13 @@ class optional
     /// @param[in] value lvalue of type T which will be copy constructed into the optional
     optional(const T& value) noexcept;
 
+    /// @brief Creates an optional and an object inside the optional on construction by perfectly forwarding args to the
+    /// constructor of T. Could be used e.g. when T is not copyable/movable.
+    /// @tparam Targs is the template parameter pack for the perfectly forwarded arguments
+    /// @param[in] Emplace_t compile time variable to distinguish between constructors with certain behavior
+    template <typename... Targs>
+    optional(Emplace_t, Targs&&... args) noexcept;
```

Comment:
Why not copying the STL and create a `cxx::make_optional` free function factory?

Reply:
... or using `cxx::in_place_t` to keep the familiarity with the STL?

Reply:
@elBoberido `make_optional` could not be used when copy/move is deleted. In this case we need this constructor. I will replace `Emplace_t` with `in_place_t`.